src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapper.java

```diff
@@ -13,20 +14,45 @@
     private static final String SDS_USER_SYSTEM = "https://fhir.nhs.uk/Id/sds-user-id";
 
     public Optional<Practitioner> mapRequester(final Message message) {
-        return message.getInvolvedParties().stream()
+        Optional<RequesterNameAndAddress> requester = message.getInvolvedParties().stream()
             .map(InvolvedParty::getRequesterNameAndAddress)
             .flatMap(Optional::stream)
-            .map(this::toPractitioner)
             .findAny();
+
+        if (requester.isPresent()) {
+            final var r = requester.get();
+            Practitioner practitioner = mapToPractitioner(r.getIdentifier(), r.getRequesterName());
+            return Optional.of(practitioner);
+        } else {
+            return Optional.empty();
+        }
     }
 
-    private Practitioner toPractitioner(final RequesterNameAndAddress requester) {
+
+    public Optional<Practitioner> mapPerformer(final Message message) {
+        Optional<PerformerNameAndAddress> performer = message.getInvolvedParties().stream()
+            .map(InvolvedParty::getPerformerNameAndAddress)
+            .flatMap(Optional::stream)
+            .findAny();
+
+        if (performer.isPresent()) {
+            final var p = performer.get();
+            Practitioner practitioner = mapToPractitioner(p.getIdentifier(), p.getPerformerName());
+            return Optional.of(practitioner);
+        } else {
+            return Optional.empty();
+        }
```

Comment:
This is all a single chain
```suggestion
        return message.getInvolvedParties().stream()
            .map(InvolvedParty::getPerformerNameAndAddress)
            .flatMap(Optional::stream)
            .map(p -> mapToPractitioner(p.getIdentifier(), p.getPerformerName())
            .findAny();
```

Reply:
Done in 5b7f289